### PR TITLE
added connection string logic

### PR DESCRIPTION
**Pull request by katiecai**
Added connection string, username/password settings

### CODE REVIEW

File: test/WebJobs.Extensions.RabbitMQ.Samples/WebJobs.Extensions.RabbitMQ.Samples.csproj [hunk 1]
```diff
@@ -30,4 +30,10 @@
     <ProjectReference Include="..\..\src\WebJobs.Extensions.RabbitMQ.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <None Update="appsettings.json">
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
```
Comment (pragnagopa) [inline test/WebJobs.Extensions.RabbitMQ.Samples/WebJobs.Extensions.RabbitMQ.Samples.csproj]: Delete or add empty appsettings.json

File: test/WebJobs.Extensions.RabbitMQ.Samples/RabbitMQSamples.cs [hunk 2]
```diff
@@ -65,18 +59,17 @@ public static class RabbitMQSamples
...
-                QueueName = "queue")] out TestClass outputMessage,
+            [RabbitMQ(QueueName = "queue")] out TestClass outputMessage,
             ILogger logger)
         {
             outputMessage = message;
             logger.LogInformation($"RabbitMQ output binding message: {JsonConvert.SerializeObject(outputMessage)}");
         }
 
         // Trigger samples
+        // Defaults to localhost if HostName is not specified
```
Comment (pragnagopa) [inline test/WebJobs.Extensions.RabbitMQ.Samples/RabbitMQSamples.cs]: Defaults to localhost if HostName is not specified and connection string is not set in  appsettings.json

File: src/Config/RabbitMQOptions.cs [hunk 3]
```diff
@@ -11,6 +11,14 @@ public class RabbitMQOptions
...
 
+        public string UserName { get; set; }
+
+        public string Password { get; set; }
+
+        public string ConnectionString { get; set; }
+
+        public int Port { get; set; }
+
         public string Exchange { get; set; }
```
[the remark this answers is not in the document]
Comment (katiecai) [reply src/Config/RabbitMQOptions.cs]: Will delete since it's not a part of attributes anymore

File: src/Config/RabbitMQOptions.cs [hunk 4]
```diff
@@ -18,9 +18,5 @@ public class RabbitMQOptions
         public string ConnectionString { get; set; }
 
         public int Port { get; set; }
-
-        public string Exchange { get; set; }
-
-        public IBasicProperties Properties { get; set; }
```
Comment (pragnagopa) [inline src/Config/RabbitMQOptions.cs]: We should let users configure IBasicProperties . Please open an issue to address this in a separate PR.

**Pull request merged**